app/models/task.py

```diff
@@ -1,5 +1,43 @@
 from app import db
 
-
 class Task(db.Model):
-    task_id = db.Column(db.Integer, primary_key=True)
+    id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    title = db.Column(db.String, nullable=False)
+    description = db.Column(db.String, nullable=False)
+    completed_at = db.Column(db.DateTime, nullable=True, default = None)
+    goal_id = db.Column(db.Integer, db.ForeignKey("goal.id"), nullable=True)
+    goal = db.relationship("Goal", back_populates="tasks")
+
+
+
+    @classmethod
+    def from_dict(cls, data_dict):
+        if "completed_at" in data_dict:
+            return cls(title = data_dict["title"], 
+            description = data_dict["description"],
+            completed_at = data_dict["completed_at"])
+        else:
+            return cls(title = data_dict["title"], 
+            description = data_dict["description"])
+
+    
+    def to_dict(self):
+        if self.completed_at:
+            flag = True
+        else:
+            flag = False
+        if self.goal_id:
```

Comment:
A slightly more succinct way to write this would be something like 
```python
result = dict(
    id = self.id,
    title = self.title,
    description = self.description,
    is_complete = flag
)

return {**result, "goal_id": self.goal_id} if self.goal_id else result
```
That uses a ternary to check for `self.goal_id`, then if it exists, uses `**` which in this context unpacks all they key/values in `result` and adds the additional `"goal_id": self.goal_id` key/value pair to it.